api/v1alpha1/capiprovider_webhook.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright SUSE 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// log is for logging in this package.
+var capiproviderlog = logf.Log.WithName("capiprovider-resource")
+
+// SetupWebhookWithManager will setup the manager to manage the webhooks
+func (r *CAPIProvider) SetupWebhookWithManager(mgr ctrl.Manager) error {
```

Comment:
I think we don’t need webhooks, and instead we can rely on kubebuilder validation or defaulting

Reply:
I just generated them as standard. So i agree if we can cover everything via kubebuilder annotations then we can remove and add again if we need them in the future.